test/ui/views/grades_view_test.dart

```diff
@@ -0,0 +1,192 @@
+
+        await expectLater(find.byType(GradesView),
+            matchesGoldenFile(goldenFilePath("gradesView_1")));
+      });
+
+      testWidgets("Multiples sessions and grades loaded",
+          (WidgetTester tester) async {
+        // Mock the repository to have 0 courses available
+        CourseRepositoryMock.stubCourses(
+            courseRepository as CourseRepositoryMock,
+            toReturn: courses);
+        CourseRepositoryMock.stubGetCourses(
+            courseRepository as CourseRepositoryMock,
+            fromCacheOnly: false);
+        CourseRepositoryMock.stubGetCourses(
+            courseRepository as CourseRepositoryMock,
+            toReturn: courses,
+            fromCacheOnly: true);
+
+        tester.binding.window.physicalSizeTestValue = const Size(800, 1410);
+
+        await tester.pumpWidget(localizedWidget(child: GradesView()));
+        await tester.pumpAndSettle();
+
+        await expectLater(find.byType(GradesView),
+            matchesGoldenFile(goldenFilePath("gradesView_2")));
+      });
+    });
+
+    group("UI -", () {
+      testWidgets(
+          "Right message is displayed when there is no grades available",
+          (WidgetTester tester) async {
+        // Mock the repository to have 0 courses available
+        CourseRepositoryMock.stubCourses(
+            courseRepository as CourseRepositoryMock);
+        CourseRepositoryMock.stubGetCourses(
+            courseRepository as CourseRepositoryMock,
+            fromCacheOnly: false);
+        CourseRepositoryMock.stubGetCourses(
+            courseRepository as CourseRepositoryMock,
+            fromCacheOnly: true);
+
+        tester.binding.window.physicalSizeTestValue = const Size(800, 1410);
+
+        await tester.pumpWidget(localizedWidget(child: GradesView()));
+        await tester.pumpAndSettle();
+
+        expect(find.text(intl.grades_msg_no_grades), findsOneWidget);
+      });
+
+      testWidgets(
+          "Correct number of grade button and right session name are displayed",
+          (WidgetTester tester) async {
+        // Mock the repository to have 4 courses available
+        CourseRepositoryMock.stubCourses(
+            courseRepository as CourseRepositoryMock,
+            toReturn: courses);
+        CourseRepositoryMock.stubGetCourses(
+            courseRepository as CourseRepositoryMock,
+            fromCacheOnly: false);
+        CourseRepositoryMock.stubGetCourses(
+            courseRepository as CourseRepositoryMock,
+            toReturn: courses,
+            fromCacheOnly: true);
+
+        tester.binding.window.physicalSizeTestValue = const Size(800, 1410);
+
+        await tester.pumpWidget(localizedWidget(child: GradesView()));
+        await tester.pumpAndSettle();
+
+        // Check the summer session list of grades.
+        final summerSessionText = find.text("${intl.session_summer} 2020");
+        expect(summerSessionText, findsOneWidget);
+        final summerList = find
```

Comment:
Nice tests 😄 Did not know I could do this, Neat!

Reply:
Very useful! :smile: